integration-tests/src/test_loop/builder.rs

```diff
@@ -158,7 +158,19 @@ impl TestLoopBuilder {
                 num_concurrent_requests_during_catchup: 1,
             }),
         };
-        client_config.tracked_shards = Vec::new();
+
+        // Configure tracked shards.
+        // * single shard tracking for validators
+        // * all shard tracking for RPCs
+        let bp_num = genesis.config.num_block_producer_seats;
+        let cp_num = genesis.config.num_chunk_producer_seats;
+        let v_num = genesis.config.num_chunk_validator_seats;
+        let validator_num = bp_num.max(cp_num).max(v_num) as usize;
+        if idx < validator_num {
+            client_config.tracked_shards = Vec::new();
```

Comment:
Should not this also be guarded by single shard tracking feature? though I guess unless single shard tracking feature is enabled, this will have no effect.

Reply:
Is there another feature for single shard tracking? I thought it's currently the only supported mode.